fine-tune docs on resource limits in deployment strategy

tnguyen-rh opened the pull request:
- intro blurb
  - invert presentation order: go from less constrained to more constrained
  - factor "optional" blurbs from example call-outs to here
- example
  - don't bother w/ "resources is optional"; renumber accordingly
  - rewrite call-out text to describe example

@derekwaynecarr 
I have a doubt about the interpretation of "100m".  That is based on reading https://github.com/GoogleCloudPlatform/kubernetes/blob/master/docs/resources.md
only; i don't know how thin a layer OpenShift adds in terms of units and suffix processing.
Could you please scrutinize that portion in particular?

@adellape @CowboysFan 
Relatedly, if OpenShift adds anything, we should probably document that somewhere (am i missing something?); if it adds nothing, we should consider pointing to upstream.

adellape commented:
@tnguyen-rh No issues.

bfallonf commented on dev_guide/deployments.adoc:
```diff
@@ -190,32 +190,28 @@ responsibility of the strategy is to make the new deployment live using the
+A deployment is completed by a pod that consumes *resources* (memory and cpu) on a node.
+Normally, pods consume until project-default limits are reached, or use unbounded node resources if no such limits are specified.
+Another way to limit resource use is to specify resource limits as part of the deployment strategy.
+In the following example, each of `resources`, `cpu` and `memory` is optional.
```
Oxford comma?

"...each of the resources, cpu, and memory..."

tnguyen-rh commented:
@adellape @bfallonf 
Thanks for the review.  I have added the oxford comma as suggested.
@derekwaynecarr @CowboysFan @vikram-redhat 
FWIW, for appinfra-docs team, i have just added a suffix table to [wip doc "Kubernetes Configuration"](http://jenkinscat.gsslab.pnq.redhat.com:8080/job/atomic-branch-tnguyen-kc-yaml/lastSuccessfulBuild/artifact/pub_Kubernetes_Configuration/index.html).

derekwaynecarr commented on dev_guide/deployments.adoc:
```diff
@@ -190,32 +190,28 @@ responsibility of the strategy is to make the new deployment live using the
 === Deployment resources
-A deployment is completed by a pod that consumes resources on a node.
+A deployment is completed by a pod that consumes *resources* (memory and cpu) on a node.
+Normally, pods consume until project-default limits are reached, or use unbounded node resources if no such limits are specified.
```
By default, a deployment pod uses unbounded node resources.  If a project has specified default container resource limits, the deployment pod's containers will consume that amount of resource.

tnguyen-rh commented:
@derekwaynecarr 
Thanks for the review.  I have updated that section along the lines you suggested.
How does the "100m" part look?

derekwaynecarr commented:
that looks fine.

----- Original Message -----
From: "Thien-Thi Nguyen" <notifications@github.com>
To: "openshift/openshift-docs" <openshift-docs@noreply.github.com>
Cc: "Derek Carr" <decarr@redhat.com>
Sent: Wednesday, June 10, 2015 3:03:34 PM
Subject: Re: [openshift-docs] fine-tune docs on resource limits in deployment strategy (#458)

@derekwaynecarr 
Thanks for the review.  I have updated that section along the lines you suggested.
How does the "100m" part look?

---
Reply to this email directly or view it on GitHub:
https://github.com/openshift/openshift-docs/pull/458#issuecomment-110879831

tnguyen-rh commented:
@derekwaynecarr 
So, last question: If i wanted to peruse OpenShift source to investigate how thin the numbers/suffix processing is (over Kubernetes, that is), what file should i start with?
@adellape 
I will squash now.

derekwaynecarr commented:
All number parsing and processing happens here:

https://github.com/openshift/origin/blob/master/Godeps/_workspace/src/github.com/GoogleCloudPlatform/kubernetes/pkg/api/resource/quantity.go

There is nothing OpenShift specific.

----- Original Message -----
From: "Thien-Thi Nguyen" <notifications@github.com>
To: "openshift/openshift-docs" <openshift-docs@noreply.github.com>
Cc: "Derek Carr" <decarr@redhat.com>
Sent: Wednesday, June 10, 2015 3:13:02 PM
Subject: Re: [openshift-docs] fine-tune docs on resource limits in deployment strategy (#458)

@derekwaynecarr 
So, last question: If i wanted to peruse OpenShift source to investigate how thin the numbers/suffix processing is (over Kubernetes, that is), what file should i start with?
@adellape 
I will squash now.

---
Reply to this email directly or view it on GitHub:
https://github.com/openshift/openshift-docs/pull/458#issuecomment-110882310

tnguyen-rh commented:
@derekwaynecarr 
Cool.  Music to my ears!
